Subject: Test scripts stooping too low
 Config::Any allows YAML::XS, YAML::Syck and YAML and prefers YAML::XS.
 However the upstream test scripts are not allowing YAML::XS.
Forwarded: no
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-12-21

--- a/t/10simple.t
+++ b/t/10simple.t
@@ -8,11 +8,14 @@
 use lib 't/lib';
 
 BEGIN {
-    eval "use YAML::Syck ()";
+    eval "use YAML::XS ()";
     if($@) {
-        eval "use YAML ()";
+        eval "use YAML::Syck ()";
         if($@) {
-            plan skip_all => "YAML or YAML::Syck required for this test";
+            eval "use YAML ()";
+            if($@) {
+                plan skip_all => "YAML::XS or YAML or YAML::Syck required for this test";
+            }
         }
     }
 
--- a/t/11default.t
+++ b/t/11default.t
@@ -12,11 +12,14 @@
 
     @classes = qw/ MXDefaultConfigTest MXDefaultMultipleConfigsTest /;
 
-    eval "use YAML::Syck ()";
+    eval "use YAML::XS()";
     if($@) {
-        eval "use YAML ()";
+        eval "use YAML::Syck ()";
         if($@) {
-            plan skip_all => "YAML or YAML::Syck required for this test";
+            eval "use YAML ()";
+            if($@) {
+                plan skip_all => "YAML::XS or YAML or YAML::Syck required for this test";
+            }
         }
     }
 
--- a/t/multiple.t
+++ b/t/multiple.t
@@ -9,11 +9,14 @@
 
 BEGIN {
 
-    eval "use YAML::Syck ()";
+    eval "use YAML::XS()";
     if($@) {
-        eval "use YAML ()";
+        eval "use YAML::Syck ()";
         if($@) {
-            plan skip_all => "YAML or YAML::Syck required for this test";
+            eval "use YAML ()";
+            if($@) {
+                plan skip_all => "YAML::XS or YAML or YAML::Syck required for this test";
+            }
         }
     }
 
